### PR TITLE
sim-model: Enable `coreneuron` variant by default in `neurodamus` packages

### PR DESCRIPTION
**Pull request by iomaganaris**

### CODE REVIEW

**Comment by iomaganaris** [comment]
> There are one or more issues with the commit message of commit [9a71e79](https://github.com/BlueBrain/spack/commit/9a71e7944f8567f29ab953c674938cff0710d71e). Commit message:
> 
> > Enable coreneuron variant by default in neurodamus packages
> 
> Please satisfy at least one of the checks (one package, docs, or deploy). The PR title must be compliant. The commit message of your commit must be compliant as well. Issues:
> 
> * The following packages were changed but not mentioned:
>   `sim-model:`
>   You can simply use the above list,                   then explain what you changed.
>   Alternatively, you can use one line per package                   to describe the change per package.
>   Please mention at least one package.
> * Deploy files were changed but not mentioned.
>   Please use the `deploy:` prefix to explain this change.

Maybe this is a bit too much? :upside_down_face:

**Review by pramodk — DISMISSED** [review]
LGTM

File: bluebrain/deployment/environments/applications_hpc.yaml [hunk 1]
```diff
@@ -32,20 +32,20 @@ spack:
           +ngv+metabolism: '{name}-multiscale/{version}'
           ^petsc+complex: '{name}-complex/{version}'
   specs:
-    - neuron+tests+coreneuron
```
Comment (ohm314) [inline bluebrain/deployment/environments/applications_hpc.yaml]: maybe not necessary to remove teh `+coreneuron` here. It doesn't hurt and at the same time it avoid confusion to keep it explicit.

**Review by ohm314 — DISMISSED** [review]

File: bluebrain/deployment/environments/applications_hpc.yaml (hunk 1 above)
Comment (iomaganaris) [reply bluebrain/deployment/environments/applications_hpc.yaml]: I see your point.. I reverted these changes

**Review by matz-e — APPROVED** [review]

**Pull request merged**